compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp

```diff
@@ -310,12 +310,14 @@ static size_t getLongestEntryLength(ArrayRef<KV> Table) {
 extern "C" void LLVMRustPrintTargetCPUs(LLVMTargetMachineRef TM) {
   const TargetMachine *Target = unwrap(TM);
   const MCSubtargetInfo *MCInfo = Target->getMCSubtargetInfo();
-  const Triple::ArchType HostArch = Triple(sys::getProcessTriple()).getArch();
+  const Triple::ArchType HostArch = Triple(sys::getDefaultTargetTriple()).getArch();
   const Triple::ArchType TargetArch = Target->getTargetTriple().getArch();
   const ArrayRef<SubtargetSubTypeKV> CPUTable = MCInfo->getCPUTable();
   unsigned MaxCPULen = getLongestEntryLength(CPUTable);
 
   printf("Available CPUs for this target:\n");
+  // Don't print the "native" entry when the user specifies --target since
```

Comment:
But we're not filtering on *any* `--target`, only if the arch doesn't match... Do we really want to show a native arch when compiling for the same arch on a different OS?

Reply:
Oh, hm, good question. I think it is probably fine just checking the arch.  If I am on x86-64 linux, and do `rustc --print=target-cpus --target x86_64-unknown-freebsd`, then showing the native entry should be fine, since the CPU list is independent of the OS, right?

(I can clarify the comment if that makes sense.)

Reply:
AFAIK the CPU list is totally independent. It's still a bit weird to use native when you're cross compiling, but who knows, maybe they're dual-booting. Anyway, let's clarify something like "--target with a different arch ..."

Reply:
Sounds good!  I agree it is a bit weird, though not entirely clear to me on how to make it better. 

For reference, this stems from [this comment](https://github.com/rust-lang/rust/pull/44407#issuecomment-329248791).